### PR TITLE
TODO: need to add support for these fields (#87, #90)

### PR DESCRIPTION
**Pull request by justin13601**
## Summary by CodeRabbit

- **New Features**
	- Enhanced admission criteria to recognize multiple event types for improved prediction capabilities in in-hospital mortality.
	- Introduced a new field to capture additional criteria, increasing the flexibility of the model.
- **Bug Fixes**
	- Improved robustness of data querying by logging warnings when no valid entries are found, enhancing error handling and clarity.
	- Ensured type consistency for the "code" column in the data processing, preventing potential errors in downstream operations.

### CODE REVIEW

**Comment by justin13601** [comment]
ADD DOCUMENTATION

**Comment by justin13601** [comment]
STILL NEED TO UPDATE DOCUMENTATION FOR THESE FUNCTIONALITY

File: sample_configs/inhospital_mortality.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Task: 24-hour In-hospital Mortality Prediction
 predicates:
   admission:
-    code: event_type//ADMISSION
+    code:
+      regex: ^event_type.*
```
Comment (mmcdermott) [inline sample_configs/inhospital_mortality.yaml:5]: ADMISSION part of the regex?

**Comment by justin13601** [comment]
@mmcdermott - please confirm if these are reasonable for the ESGPT version of #90 :

- For ESGPT regex and any - what is the functionality we want to support? `BP//systolic` refers to `systolic` in `BP`; so would we want `BP//<regex>` to match regex in column `BP`?

- What about `any`? Mandate users enter `{BP: ['val1', 'val2', 'val3']}` for the field `any` under `code`? Wonder if that is too complicated?

- There is still `event_type`, which I guess could be similar to above and just have `event_type//<regex>` or `{event_type: ['val1', 'val2', 'val3']}` after splitting by `&`?

- Finally, there is having codes not in parts: `BP`, which refers to `col("BP").is_not_null()`, do we want regex and any for multiple columns?

File: src/aces/config.py [hunk 2]
```diff
@@ -60,19 +61,92 @@ def MEDS_eval_expr(self) -> pl.Expr:
...
+                    criteria.append(pl.col("numerical_value") >= self.value_min)
+                else:
+                    criteria.append(pl.col("numerical_value") > self.value_min)
+            if self.value_max is not None:
+                if self.value_max_inclusive:
+                    criteria.append(pl.col("numerical_value") <= self.value_max)
+                else:
+                    criteria.append(pl.col("numerical_value") < self.value_max)
+
+        if self.other_cols:
```
[the remark this answers is not in the document]
Comment (mmcdermott) [reply src/aces/config.py:148]: This is actually a good suggestion, but we can implement this sort of cleanup later.

**Review by mmcdermott — APPROVED** [review]

**Pull request merged**